airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -139,20 +152,9 @@ def _get_branches_data(selected_branches: str, full_refresh_args: Dict[str, Any]
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         try:
             authenticator = self._get_authenticator(config)
-            # In case of getting repository list for given organization was
-            # successfull no need of checking stats for every repository within
-            # that organization.
-            # Since we have "repo" scope requested it should grant access to private repos as well:
-            # https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes
-            repositories, _ = self._generate_repositories(config=config, authenticator=authenticator)
-
-            repository_stats_stream = RepositoryStats(
-                authenticator=authenticator,
-                repositories=repositories,
-                page_size_for_large_streams=config.get("page_size_for_large_streams", DEFAULT_PAGE_SIZE_FOR_LARGE_STREAM),
-            )
-            for stream_slice in repository_stats_stream.stream_slices(sync_mode=SyncMode.full_refresh):
-                next(repository_stats_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
+            organizations, repositories = self._get_org_repositories(config=config, authenticator=authenticator)
+            if not repositories:
```

Comment:
Is a config considered valid if it returns at least one organization, but no repositories?

And if we actually don't need to check organizations in the `check()` method, can we ignore the first part of the tuple in the above like: `_, repositories = self._get_org_repositories(config=config, authenticator=authenticator)`

Reply:
> Is a config considered valid if it returns at least one organization, but no repositories?

config is valid if we have at least one repository

> can we ignore the first part

good catch - fixed, thank you